app/lib/activitypub/activity/create.rb

```diff
@@ -402,15 +406,17 @@ def addresses_local_accounts?
 
     return false if local_usernames.empty?
 
-    Account.local.where(username: local_usernames).exists?
+    scope = Account.local.where(Account.arel_table[:username].lower.in(local_usernames.map(&:downcase)))
+    scope = scope.where.not('EXISTS (SELECT 1 FROM blocks WHERE account_id = accounts.id AND target_account_id = ?)', @account.id)
+    scope.exists?
```

Comment:
There is a slight (probably pretty much unobservable) overhead to checking for blocks, but the overall query is actually much faster by avoiding a sequential scan (`Account.local.where(username: local_usernames)` did not use the index we have, because the index is on lowered usernames).